sdk/iot/hub/samples/src/sample_base64.h

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_result.h"
+#include "az_span.h"
+
+/*
+ * This serves as an example for fundamental functionality needed to use SAS key authentication.
+ * This implementation uses OpenSSL.
+ */
+
+// Decode an input span from base64 to raw bytes
+az_result sample_base64_decode(az_span base64_encoded, az_span in_span, az_span* out_span);
+
+// Encode an input span of raw bytes to base64
+az_result sample_base64_encode(az_span bytes, az_span in_span, az_span* out_span);
+
+// HMAC256 an input span with an input key
+az_result sample_hmac_sha256_encrypt(az_span key, az_span bytes, az_span in_span, az_span* out_span);
```

Comment:
>sample_hmac_sha256_encrypt [](start = 10, length = 26)

this file is doing more then base64... can we name it something else? Or do we even need it.

Reply:
I can rename it. I would like to keep it as a separate file because we will likely have other samples with SAS key auth and this prevents duplicating code.
How about `sample_sas_gen_helper.h` or something like that.

Reply:
Utility functions.?

---
In reply to: [436076986](https://github.com/Azure/azure-sdk-for-c/pull/808#discussion_r436076986) [](ancestors = 436076986)

Reply:
I'd like to keep sas in there so how about 
`sample_sas_utility.h`

Reply:
One thing to think about is if we only want to share sas functions between samples then I it should be fine, but if we have bigger scope for this file then adding sas_utitlity might not make sesnse?

---
In reply to: [436085582](https://github.com/Azure/azure-sdk-for-c/pull/808#discussion_r436085582) [](ancestors = 436085582)

Reply:
The base64 encoding and decoding should be handled by the user. Same with the hmac256. So here, it is useful in making explicit what functionality would need to be implemented by the user in order to get the SAS functionality to work. The scope should be kept only to these samples since our public API's shouldn't touch/involve themselves with these. 

I believe there is broader effort to consolidate reused sample code into common source files so that we don't have to maintain multiple instances of the same thing. This aligns with that.